### PR TITLE
Perform recursive validation.

### PR DESCRIPTION
**Pull request by StephaneAlie**

### CODE REVIEW

File: src/Georadix.Core.Tests/Validation/DataAnnotationExtensionsFixture.cs [hunk 1]
```diff
@@ -24,43 +45,64 @@ public void AssertValidOnInvalidObjectThrowsArgumentException()
...
         }
 
-        private class TestModel
+        public class SubModel
         {
+            [Required(AllowEmptyStrings = false, ErrorMessage = "Value is required.")]
+            public string Property1 { get; set; }
+        }
+
+        public class TestModel
```
Comment (AnthonyMacKinnon) [inline src/Georadix.Core.Tests/Validation/DataAnnotationExtensionsFixture.cs]: Do these classes need to be public?

**Comment by AnthonyMacKinnon** [comment]
LGTM

**Comment by dalie** [comment]
LGTM

File: src/Georadix.Core.Tests/Validation/DataAnnotationExtensionsFixture.cs (hunk 1 above)
Comment (StephaneAlie) [inline src/Georadix.Core.Tests/Validation/DataAnnotationExtensionsFixture.cs]: They had to once they got exposed in the test method as a parameter, it was complaining that the access modified for the class was more restrictive than the method.

**Pull request merged**